cmd/promtool/archive.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"os"
+)
+
+const filePerm = 0644
+
+type archiverConfig struct {
+	archiveName string
+}
+
+type archiver interface {
+	Write(fileName string, buf *bytes.Buffer) error
+	Close() error
+	File() *os.File
```

Comment:
These aren't used outside of this package (and can't be, since the surrounding type is unexported), so unexport (lower-case) them.

Reply:
@krasi-georgiev if we make the `File()` unexported, what new name for the `file` field in `tarGzFileWriter` should we use?

Reply:
@juliusv the problem here was the we can't have a method `file`  and a field name `file` in  `tarGzFileWriter`

Reply:
That's not a great reason for changing the exposure of a method though... actually I don't think you even need the `File()` method on here, because outside of the type you don't even need the `*os.File`, you just need the filename, so it could be `filename()`? (don't expose more than you have to)